### DISCUSSION TITLE
Contrast too feint

### DISCUSSION THREAD

**Discussion by cobusvrooyen** [olikraus/u8g2#1836] [Q&A]
This is my first tussle with a 12864I LCD screen and ESP32s combination. 

I am using Arduino IDE and various u8g2 examples but am am struggling to get a crisp display. 

Looking at the screen at a right angle it appears as if nothing is displayed but when looking from a sideways angle it is possible to see the output. I tried the contrast example code to see if it can get any better but there's hardly a difference as it steps through the various modes.

I could not find a lot of information on the interwebs for my hardware config (ESP32S and 12864I). These two sites below was the most helpful for me. Should I rather try another type of MCU?
[https://forum.arduino.cc/t/how-to-connect-lcd-12864-st7920-to-nodemcu-esp8266-module/534077/2](url)
[https://thesolaruniverse.wordpress.com/2020/06/09/esp8266-and-st7920-powered-12864-liquid-crystal-display-in-a-test-bench/](url)

![image](https://user-images.githubusercontent.com/72613128/162819216-e39f2c6c-51a8-4ef4-ac97-0856f7934309.png)

**Comment by olikraus** [comment]
If I remember correctly, you need a varpot to adjust the contranst:
https://content.instructables.com/ORIG/F6G/FDL5/K9317R6C/F6GFDL5K9317R6C.jpg?auto=webp&frame=1&width=614&fit=bounds&md=b3ad679af75f2648f42d375655139934

**Comment by cobusvrooyen** [comment]
That's it! I recall seeing a pot in one of the write-ups but it slipped my mind again. Thanks!
